### PR TITLE
gh-124295: Add translation tests for argparse

### LINKED ISSUE #124295: Add translation tests for argparse
While #12711 was merged, it seems like there's a separate issue worth tracking around adding translation tests for argparse (see https://github.com/python/cpython/pull/12711#issuecomment-1964754284)

### Linked PRs
* gh-124803
* gh-126046
* gh-126051
* gh-126054

### PR DESCRIPTION
**Pull request by tomasr8**
This uses `Tools/i18n/pygettext.py` to extract all translatable messages from `argparse.py`. The messages are kept in a snapshot file in the test folder. The test extracts the current messages and compares them against this snapshot.

The snapshot can be updated using `./python Lib/test/test_argparse/test_translations.py --snapshot-update`.

There is currently no way to read a PO file using the stdlib (gettext only reads MO files, maybe it should be able to do both?) so I just use a simple regex to get the `msgid`s out of the PO file.

Feedback welcome!

* Issue: gh-124295

### CODE REVIEW

File: Lib/test/test_argparse/test_translations.py
Comment (savannahostrowski) [inline Lib/test/test_argparse/test_translations.py:1]: Does this need to be in a new file? Could we not just add these tests to the existing test_argparse.py file?

File: Lib/test/test_argparse/test_translations.py
Comment (serhiy-storchaka) [inline Lib/test/test_argparse/test_translations.py]: This does not work in the case like: ``` "abc\\"def" ```.
```suggestion
msgid_string_pattern = re.compile(r'"((?:\\.|[^"])*)"')
```

Comment (serhiy-storchaka) [reply Lib/test/test_argparse/test_translations.py]: On other hand, this should never happen in the PO files. So it is fine as is.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM. Thank you, @tomasr8.

Since argparse is currently actively fixed, there are some conflicts. After fixing them I'll merge this PR.

**Comment by serhiy-storchaka** [comment]
I will wait for the 3.13 release before merging this PR because it will conflict with many backports waiting for 3.13. It would be easier to resolve conflicts in this PR (just move `test_argparse.py`) than in other PRs which modify the part of the file.

**Comment by tomasr8** [comment]
> I will wait for the 3.13 release before merging this PR because it will conflict with many backports waiting for 3.13. It would be easier to resolve conflicts in this PR (just move `test_argparse.py`) than in other PRs which modify the part of the file.

Ok! Let me know if I can help with anything :)

**Comment by serhiy-storchaka** [comment]
Don't worry, I'll merge this PR when the time comes.

**Comment by tomasr8** [comment]
> Don't worry, I'll merge this PR when the time comes.

Github was complaining about some merge conflicts so I just wanted to update the branch ;)

**Review by erlend-aasland — APPROVED** [review]

**Comment by tomasr8** [comment]
On Serhiy's suggestion, I moved the translation data into a separate folder and merged the new tests with the existing ones to avoid turning the tests into a package.

File: Lib/test/test_argparse.py
Comment (serhiy-storchaka) [inline Lib/test/test_argparse.py]: You can use `REPO_ROOT` and `TEST_HOME_DIR`  from `test.support`.

File: Lib/test/test_argparse.py
Comment (serhiy-storchaka) [inline Lib/test/test_argparse.py]: You can simply use `requires_subprocess`.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by tomasr8** [comment]
There seems to be a buildbot failure related to this PR: https://buildbot.python.org/#/builders/1244/builds/3363
I think it's because scripts are not always included in python installations. We'll probably need to guard the test with `skip_if_missing`.

**Comment by serhiy-storchaka** [comment]
Do you mind to create a PR? I paused backporting.

Simply call `skip_if_missing()` in the test. Or in `classSetUp()` if there were several tests.

**Comment by tomasr8** [comment]
> Do you mind to create a PR? I paused backporting.
> 
> Simply call `skip_if_missing()` in the test. Or in `setUpClass()` if there were several tests.

On it :)

**Comment by tomasr8** [comment]
PR is here: https://github.com/python/cpython/pull/126051

**Pull request merged**
